src/Context/ThemeContext.tsx

```diff
@@ -0,0 +1,33 @@
+
+import { createContext, useState , ReactNode} from 'react';
+
+export interface ITheme{
+    theme: string;
+    setTheme: React.Dispatch<React.SetStateAction<string>>;
+    toggleTheme: () => void;
+}
+interface ThemeContextProviderProps {
+    children: ReactNode;
+  }
+export const ThemeContext = createContext<ITheme | null>(null);
+
+const ThemeContextProvider: React.FC<ThemeContextProviderProps>= (props)=>{
+    const [theme, setTheme] = useState('light');
```

Comment:
for better user experience, you can store the theme in the local storage and retrieve it as the default value

`const [theme, setTheme] = useState(()=> localStorage.getItem("theme") || "light");`

Reply:
done